src/main.ts

```diff
@@ -124,26 +125,47 @@ export default class VaultChat extends Plugin {
 	onunload() {
 	}
 
-	async searchForTerm(searchTerm: string): Promise<Array<string>> {
-		if (searchTerm === '') {
-			return []
-		}
-		const embedding = await this.openAIHandler.createEmbedding(searchTerm)
-		if (embedding === undefined) {
-			console.error(`Failed to generate vector for search term.`)
-			return []
+	async loadSettings() {
+		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
+	}
+
+	async saveSettings() {
+		await this.saveData(this.settings);
+		if (this.waitingForApiKey && this.apiKeyIsValid()) {
+			this.waitingForApiKey = false
+			await this.initializePlugin()
 		}
-		const nearest: NearestVectorResult[] = this.vectorStore.getNearestVectors(embedding, 3, this.settings.relevanceThreshold)
-		return nearest.map(n => n.path)
 	}
 
-	async getSearchResultsFiles(searchTerm: string): Promise<Array<SearchResult>> {
-		const embeddingResponse = await this.openAIHandler.createEmbedding(searchTerm)
-		if (embeddingResponse === undefined) {
-			console.error(`Failed to generate vector for search term.`)
-			return []
+	async askChatGpt(question: string) {
+		// HyDE: request note that answers the question
+		const conversation: Array<ChatCompletionRequestMessage> = []
+		const hydeMessage: ChatCompletionRequestMessage = {
+			role: ChatCompletionResponseMessageRoleEnum.User,
+			content: `${hydePrompt} ${question}`,
 		}
-		const nearestVectors = this.vectorStore.getNearestVectors(embeddingResponse, 8, this.settings.relevanceThreshold)
+		conversation.push(hydeMessage)
+		const hydeResponse = await this.openAIHandler.createChatCompletion(conversation)
+		if (!hydeResponse || hydeResponse.choices.length === 0 || !hydeResponse.choices[0].message) {
```

Comment:
Improvement idea: instead of failing here, let's degrade and just search based on your search without the hyde.  Does that make senser?  Shouldn't be this PR

Reply:
Good idea! I agree. We should make this degrade to simple embeddings search. (Though in your example if completions was completely down the user wouldn't be successfully asking GPT any questions either.)

Reply:
true this would only apply to semantic search